Source/WebCore/platform/graphics/texmap/ClipPath.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2024 Jani Hautakangas <jani@kodegood.com>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following
+ *    disclaimer in the documentation and/or other materials provided
+ *    with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "ClipPath.h"
+
+#include <wtf/TZoneMallocInlines.h>
+
+namespace WebCore {
+
+WTF_MAKE_TZONE_ALLOCATED_IMPL(ClipPath);
+
+ClipPath::ClipPath(Vector<FloatPoint>&& vertices, unsigned bufferID, unsigned bufferOffsetInBytes)
+    : m_vertices(WTFMove(vertices))
+    , m_bufferID(bufferID)
+    , m_bufferOffsetInBytes(bufferOffsetInBytes)
+{
+    unsigned vertexCount = numberOfVertices();
+    if (vertexCount) {
+        m_bounds.setLocation(m_vertices.at(0));
+        for (size_t i = 1; i < vertexCount; i++)
+            m_bounds.extend(m_vertices.at(i));
+    }
+}
+
+const void* ClipPath::data() const
+{
+    if (m_bufferID)
+        return reinterpret_cast<const void*>(m_bufferOffsetInBytes);
+
+    return m_vertices.data();
```

Comment:
Why do you return m_vertices.data() here? When is m_bufferID is 0?

Reply:
At some point I had a plan that ClipPath could be used CPU buffer for clip vertices and then uploaded to GPU on demand. I did proceed with that approach and this is left over from that. Removed now and function name updated to `bufferDataOffsetAsPtr()`